modules/node/nodes.tf

```diff
@@ -0,0 +1,55 @@
+# public ip address for scaleway instance
+resource "scaleway_ip" "node_ip" {
+  count = "${var.instance_count}"
+}
+
+# scaleway instance
+resource "scaleway_server" "instance" {
```

Comment:
You may want to use the new resource [`scaleway_instance_server`](https://www.terraform.io/docs/providers/scaleway/r/instance_server.html) which comes with [the placement group feature](https://www.terraform.io/docs/providers/scaleway/r/instance_placement_group.html)